python/pyspark/sql/tests/test_context.py

```diff
@@ -40,15 +40,20 @@ def setUpClass(cls):
         ReusedPySparkTestCase.setUpClass()
         cls.tempdir = tempfile.NamedTemporaryFile(delete=False)
         cls.hive_available = True
+        cls.spark = None
         try:
             cls.sc._jvm.org.apache.hadoop.hive.conf.HiveConf()
```

Comment:
@srowen, seems this code is still needed. `SparkSession.builder.enableHiveSupport().getOrCreate()` seems not able to directly check if Hive is available or not. I manually tested and just pushed the changes directly into your branch.